sklearn/preprocessing/_data.py

```diff
@@ -776,10 +775,21 @@ def partial_fit(self, X, y=None, sample_weight=None):
         # if n_samples_seen_ is an integer (i.e. no missing values), we need to
         # transform it to a NumPy array of shape (n_features,) required by
         # incr_mean_variance_axis and _incremental_variance_axis
-        if (hasattr(self, 'n_samples_seen_') and
-                isinstance(self.n_samples_seen_, numbers.Integral)):
+        if not hasattr(self, 'n_samples_seen_'):
```

Comment:
```suggestion
        dtype = np.int64 if sample_weight is None else X.dtype
        if not hasattr(self, 'n_samples_seen_'):
```

Reply:
I would rather push the `dtype = np.int64 if sample_weight is None else X.dtype` line inside the `if not hasattr(self, 'n_samples_seen_'):` block as it's not used outside that block.

Reply:
it's used now outside. Will push in a minute